src/lib.rs

```diff
@@ -159,6 +204,21 @@ fn test_pop_push() {
     assert_eq!([1], *a);
 }
 
+#[test]
+fn test_update() {
+    let mut a = vec![1];
+    {
+        assert_eq!([-1], *a.update(0, -1));
+    }
+    assert_eq!([1], *a);
+
+    let mut a = vec![0, 1, 2, 3];
```

Comment:
To me, this second test is covering the same use case as the first test.

Reply:
good point, i'll remove the first one 1b8406697f8dad4681abe9df21010d11dc8e53c5